climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1350 @@
+        # assign new event IDs
+        event_id_new = 1
+        for i in np.unique(firms.event_id):
+            firms.event_id[firms.event_id == i] = event_id_new
+            event_id_new = event_id_new + 1
+
+        firms = firms.reset_index()
+
+        return firms
+
+    def _calc_brightness(self, firms, centroids, res_centr):
+        """ Compute intensity matrix per fire with the maximum brightness at
+        each centroid and all other hazard attributes.
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+            centroids : Centroids
+            res_centr : float
+                centroids resolution in centroids unit
+
+        Returns
+        -------
+            self : climada.hazard.WildFire instance
+
+        """
+        uni_ev = np.unique(firms['event_id'].values)
+        num_ev = uni_ev.size
+        num_centr = centroids.size
+
+        # For one fire, if more than one points of firms dataframe have the
+        # same coordinates, take the maximum brightness value
+        # of these points (maximal damages).
+        tree_centr = BallTree(centroids.coord, metric='chebyshev')
+        if self.pool:
+            chunksize = min(num_ev//self.pool.ncpus, 1000)
+            bright_list = self.pool.map(self._brightness_one_fire,
+                                        itertools.repeat(firms, num_ev),
+                                        itertools.repeat(tree_centr, num_ev),
+                                        uni_ev, itertools.repeat(res_centr),
+                                        itertools.repeat(num_centr),
+                                        chunksize=chunksize)
+        else:
+            bright_list = []
+            for ev_id in uni_ev:
+                bright_list.append(self._brightness_one_fire(firms, tree_centr, \
+                ev_id, res_centr, num_centr))
+
+        bright_list, firms = self._remove_empty_fires(bright_list, firms)
+
+        uni_ev = np.unique(firms['event_id'].values)
+        num_ev = uni_ev.size
+
+        # save
+        self.tag = TagHazard('WFsingle')
+        self.centroids = centroids
+        self.units = 'K' # Kelvin brightness
+
+        # Following values are defined for each fire
+        self.event_id = np.arange(1, num_ev+1).astype(int)
+        self.event_name = list(map(str, self.event_id))
+        self.date = np.zeros(num_ev, int)
+        self.date_end = np.zeros(num_ev, int)
+        for ev_idx, ev_id in enumerate(uni_ev):
+            self.date[ev_idx] = firms[firms.event_id == ev_id].datenum.min()
+            self.date_end[ev_idx] = firms[firms.event_id == ev_id].datenum.max()
+        self.orig = np.ones(num_ev, bool)
+        self._set_frequency()
+
+        # Following values are defined for each fire and centroid
+        self.intensity = sparse.lil_matrix(np.zeros((num_ev, num_centr)))
+        for idx, ev_bright in enumerate(bright_list):
+            self.intensity[idx] = ev_bright
```

Comment:
Is it efficient to first make a sparse matrix, then add items to it, and then convert it to a different sparse matrix? I think `bright_list`could directly be converted to a sparse matrix of a pre-determined shape?

Reply:
cf https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html